whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/processor/UpdateNotificationFileProcessor.java

```diff
@@ -48,24 +47,30 @@ public class UpdateNotificationFileProcessor {
 
     private final static String PUBLIC_KEY_PATH = "public.key";
 
+    private final String baseUrl;
 
-    public UpdateNotificationFileProcessor(final NrtmRestClient nrtmRestClient,
-                                           final Nrtm4ClientInfoRepository nrtm4ClientMirrorDao,
-                                           final SnapshotMirrorImporter snapshotImporter,
-                                           final DeltaMirrorImporter deltaImporter) {
+    public UpdateNotificationFileProcessor(
+            @Value("${nrtm.baseUrl}") final String baseUrl,
+            final NrtmRestClient nrtmRestClient,
+            final Nrtm4ClientInfoRepository nrtm4ClientMirrorDao,
+            final SnapshotMirrorImporter snapshotImporter,
+            final DeltaMirrorImporter deltaImporter) {
         this.nrtmRestClient = nrtmRestClient;
         this.nrtm4ClientMirrorDao = nrtm4ClientMirrorDao;
         this.snapshotImporter = snapshotImporter;
         this.deltaImporter = deltaImporter;
+        this.baseUrl = baseUrl;
     }
 
     public void processFile(){
+        final StringBuilder filesCommonPath = new StringBuilder(baseUrl); // unf, snap, delta common path of their URL (match the RFC)
```

Comment:
Can you use something similar ?
URL url=new URL("https://nrtm.db.ripe.net/nrtmv4/RIPE-NONAUTH/update-notification-file.json ");
        final String relativePath = "nrtm-delta.2741.RIPE-NONAUTH.4f3ff2a7-1877-4cab-82f4-1dd6425c4e7d.e93c7e40830f5aa42976d6e918000e96.json";

        URL newUrl = new URL(url, relativePath);

I think we should create a relative path from UNF url instead of seperating it in a common path variable.

Reply:
`    URI getUriFromRelativePath(final URI unfUri, final String relativePath){
        return unfUri.resolve(relativePath);
    }`